rapids-4-spark-tools/src/main/scala/com/nvidia/spark/rapids/tool/profiling/ProfileMain.scala

```diff
@@ -147,4 +145,58 @@ object ProfileMain extends Logging {
 
     0
   }
+
+  /**
+   * Function to evaluate the event logs to be processed.
+   *
+   * @param filterNLogs    number of event logs to be selected
+   * @param matchlogs      keyword to match file names in the directory
+   * @param eventLogsPaths Array of event log paths
+   * @return event logs to be processed
+   */
+  def processAllPaths(
+      filterNLogs: ScallopOption[String],
+      matchlogs: ScallopOption[String],
+      eventLogsPaths: List[String]): Seq[Path] = {
+
+    val allPaths: ArrayBuffer[Path] = ArrayBuffer[Path]()
+    var allPathsWithTimestamp: Map[Path, Long] = Map.empty[Path, Long]
+
+    for (pathString <- eventLogsPaths) {
+      val paths = ProfileUtils.stringToPath(pathString)
+      if (paths.nonEmpty) {
+        allPaths ++= paths.map(a => a._1)
+        allPathsWithTimestamp ++= paths
+      }
+    }
+
+    // Filter the eventlogs to be processed based on the criteria. If it is not provided in the
+    // command line, then return all the event logs processed above.
+    if (matchlogs.isDefined || filterNLogs.isDefined) {
+      if (matchlogs.isDefined) {
+        allPathsWithTimestamp =
```

Comment:
nit: (you don't need to change)
this would be more readable like this:
```
allPathsWithTimestamp.filter { case (path, _) =>
   path.getName.contains(matchlogs.toOption.get))
}
```
Note the getName instead of toString and easier to understand what is with the t._1.

Reply:
Thanks! You are right, it's more readable.